tests/ftqc/test_decomposition.py

```diff
@@ -0,0 +1,185 @@
+            qml.Rot(np.pi / 3, -0.5, np.pi / 5, 2)
+            qml.CZ([0, 1])
+            qml.CY([1, 2])
+            return qml.state()
+
+        decomposed_circuit = convert_to_mbqc_gateset(circuit)
+
+        assert np.allclose(circuit(), decomposed_circuit())
+
+    def test_explicit_mbqc_implementation_matches(self):
+        """Test that the explicit MBQC implementation of the RotXZX gate that
+        a Rot gate decomposes to produces the expected analytic result"""
+
+        dev = qml.device("default.qubit")
+
+        state = np.random.random(2) + 1j * np.random.random(2)
+        state = state / np.linalg.norm(state)
+
+        @qml.qnode(dev)
+        def rot_ref(angles):
+            """Reference circuit for applying some state preparation and then
+            performing a rotation, expressed using the PL Rot gate (ZYZ)"""
+            qml.StatePrep(state, wires=0)
+            qml.Rot(*angles, wires=0)
+            return qml.expval(qml.X(0)), qml.expval(qml.Y(0)), qml.expval(qml.Z(0))
+
+        @diagonalize_mcms
+        @qml.qnode(dev, mcm_method="tree-traversal")
+        def rot_mbqc(rot_xzx_gate):
+            """This circuit accepts a RotXZX gate, and creates a circuit equivalent
+            to performing state preparation and then applying the rotation gate,
+            with the rotation expressed in the MBQC formalism"""
+            # prep input node
+            qml.StatePrep(state, wires=[0])
+
+            # prep graph state
+            GraphStatePrep(nx.grid_graph((4,)), wires=[1, 2, 3, 4])
+
+            # entangle input and graph state
+            qml.CZ([0, 1])
+
+            # MBQC Z rotation: X, X, +/- angle, X
+            angles = rot_xzx_gate.single_qubit_rot_angles()
+            m1 = measure_x(0)
+            m2 = cond_measure(
+                m1,
+                partial(measure_arbitrary_basis, angle=angles[0]),
+                partial(measure_arbitrary_basis, angle=-angles[0]),
+            )(plane="XY", wires=1)
+            m3 = cond_measure(
+                m2,
+                partial(measure_arbitrary_basis, angle=angles[1]),
+                partial(measure_arbitrary_basis, angle=-angles[1]),
+            )(plane="XY", wires=2)
+            m4 = cond_measure(
+                (m1 + m3) % 2,
+                partial(measure_arbitrary_basis, angle=angles[2]),
+                partial(measure_arbitrary_basis, angle=-angles[2]),
+            )(plane="XY", wires=3)
+
+            # corrections based on measurement outcomes
+            qml.cond((m1 + m3) % 2, qml.Z)(4)
+            qml.cond((m2 + m4) % 2, qml.X)(4)
+
+            return qml.expval(qml.X(4)), qml.expval(qml.Y(4)), qml.expval(qml.Z(4))
+
+        angles = 1.39, -0.123, np.pi / 7
+
+        # convert Rot to XZX using the decomposition rule
+        with qml.queuing.AnnotatedQueue() as q:
+            _rot_to_xzx(*angles, wires=0)
+        xzx_op = q.queue[0]
+
+        assert np.allclose(rot_ref(angles), rot_mbqc(xzx_op))
```

Comment:
I think comparing these executions is fine for now. Maybe we can identify if a pattern-matched comparison to the expected generated gate-set (ie without execution) is a potential later addition.

Reply:
I kind of thought of it as a stand-in for an E2E integration test for decomposition - we don't have the last step in-place yet, but we want to make sure this decomposition works "all the way" to the final MBQC formalism. 

Once we have a full decomposition pipeline in place, I think the explicit circuit could be replaced with `diagonalize_mcms(convert_to_mbqc_formalism(convert_to_mbqc_gateset(rot_ref)))`. But also happy to update it to use pattern matching and have a full integration test for decomposition be a separate thing.

Reply:
Let's revisit then when the full pipeline is in place. All good for now